groups/instance/iam.tf

```diff
@@ -44,15 +44,43 @@ data "aws_iam_policy_document" "ssm_service" {
   }
 }
 
-data "aws_iam_policy" "ssm_service_core" {
-  arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+data "aws_iam_policy_document" "kms_key" {
+  statement {
+    sid       = "AllowKmsOperations"
+    effect    = "Allow"
+    resources = ["*"]
```

Comment:
I believe this would allow the EC2 instance access to any KMS key within the account.  It would be more secure to specify the artifactory kms key arn in the resources part, so that it just gives access to that one.

Reply:
artifactory-devops-kms-policy permissions: resource now restricted to the KMS Key ARN as advised. Full stop & start process completed on the instance to ensure that the EBS is still in use from decryption etc.